pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -132,30 +146,23 @@ func newManagerImpl(socketPath string) (*ManagerImpl, error) {
 	return manager, nil
 }
 
-func (m *ManagerImpl) genericDeviceUpdateCallback(resourceName string, added, updated, deleted []pluginapi.Device) {
-	kept := append(updated, added...)
+func (m *ManagerImpl) genericDeviceUpdateCallback(resourceName string, devices []pluginapi.Device) {
 	m.mutex.Lock()
-	if _, ok := m.healthyDevices[resourceName]; !ok {
-		m.healthyDevices[resourceName] = sets.NewString()
-	}
-	if _, ok := m.unhealthyDevices[resourceName]; !ok {
-		m.unhealthyDevices[resourceName] = sets.NewString()
-	}
-	for _, dev := range kept {
+	m.healthyDevices[resourceName] = sets.NewString()
```

Comment:
We're loosing the ability to react to a device appearing / disappearing or changing state.
That's not a problem since we're not doing anything now but it probably needs some discussion in a separate issue.

Reply:
thanks, you are right, we are not doing anything now, we can achieve this capability without endpoint cache,  just need to compare the updated devices list with `healthyDevices` + `unhealthyDevices`,  do you think we need to create an issue to track this?

Reply:
@figo +1